mirgecom/boundary.py

```diff
@@ -1370,114 +1446,85 @@ class IsothermalWallBoundary(PrescribedFluidBoundary):
-    .. automethod:: grad_cv_bc
+    .. automethod:: state_plus_inviscid
+    .. automethod:: state_bc
     .. automethod:: temperature_bc
-    .. automethod:: isothermal_wall_state
+    .. automethod:: grad_cv_bc
     """
 
     def __init__(self, wall_temperature=300):
         """Initialize the boundary condition object."""
-        self._wall_temp = wall_temperature
         PrescribedFluidBoundary.__init__(
-            self, boundary_state_func=self.isothermal_wall_state,
-            inviscid_flux_func=self.inviscid_wall_flux,
-            viscous_flux_func=self.viscous_wall_flux,
-            boundary_temperature_func=self.temperature_bc,
-            boundary_gradient_cv_func=self.grad_cv_bc
-        )
-
-    def isothermal_wall_state(
-            self, dcoll, dd_bdry, gas_model, state_minus, **kwargs):
-        """Return state with zero-velocity and the respective internal energy."""
-        temperature_wall = self._wall_temp + 0*state_minus.mass_density
-        mom_plus = 0*state_minus.momentum_density
-        mass_frac_plus = state_minus.species_mass_fractions
+            self,
+            boundary_state_func=self.state_bc,  # For grad(CV)
+            boundary_temperature_func=self.temperature_bc,  # For grad(T)
+            boundary_gradient_cv_func=_do_not_call,
+            gradient_numerical_flux_func=_ldg_bnd_flux_for_grad,
+            inviscid_flux_func=partial(
+                _inviscid_flux_for_prescribed_state_mengaldo,
+                state_plus_func=self.state_plus_inviscid),
+            viscous_flux_func=partial(
+                _viscous_flux_for_prescribed_state_mengaldo,
+                state_bc_func=self.state_bc,
+                grad_cv_bc_func=self.grad_cv_bc,
+                grad_temperature_bc_func=None))
+
+        self._isothermal = _IsothermalBoundaryComponent(wall_temperature)
+        self._no_slip = _NoSlipBoundaryComponent()
+        self._impermeable = _ImpermeableBoundaryComponent()
+
+    def state_plus_inviscid(self, dcoll, dd_bdry, gas_model, state_minus, **kwargs):
+        r"""
+        Return state with opposite velocity.
+
+        Specialized for the inviscid operator in the sense that $(\rho E)$ is left
+        unmodified (even though $T$ is modified). This ensures that energy is not
+        advected through the wall.
+        """
+        dd_bdry = as_dofdesc(dd_bdry)
+        normal = state_minus.array_context.thaw(dcoll.normal(dd_bdry))
 
-        internal_energy_plus = gas_model.eos.get_internal_energy(
-            temperature=temperature_wall, species_mass_fractions=mass_frac_plus)
+        mom_plus = self._no_slip.momentum_plus(state_minus.momentum_density, normal)
 
-        # Velocity is pinned to 0 here, no kinetic energy
-        total_energy_plus = state_minus.mass_density*internal_energy_plus
+        # Don't modify the energy, even though we modify the temperature; energy will
+        # be advected through the wall, which doesn't make sense
```

Comment:
I hope that the flux, and any dissipation term vanishes in the energy component.  I will check the tests and add if necessary.